entrypoint.sh

```diff
@@ -1,10 +1,12 @@
 #!/usr/bin/env sh
 
-set -o errexit
-
-IDEA_JDK="/app/idea-IC/jre64"
-export IDEA_JDK
-TMPDIR="${XDG_RUNTIME_DIR}/app/${FLATPAK_ID}"
-export TMPDIR
+export JAVA_HOME=/app/jdk
+export JAVA_TOOL_OPTIONS=-Djava.io.tmpdir=${XDG_CACHE_HOME}/tmp
+export MAVEN_HOME=/app/idea-IC/plugins/maven/lib/maven3
```

Comment:
This won't work here; this is usually used by Maven 1.x as `M2_HOME` is for Maven 2.x

There are a plethora of build tools and plugins that will use the default location(s). You would have to overwrite all of that with time.

Reply:
Today I actually double checked this again and M2_HOME env variable usage has been dropped since Maven > 3.5. For what I understood MAVEN_HOME is the only one that should be used.
The integrated terminal in IntelliJ works fine with this setup and also the .m2 folder is stored in /var/data/.m2/repository/.
Please let me know if I can test other plugins to confirm they don't break.

Reply:
That's the problem: I don't know how many plugins might not behave correctly :rofl:

There are tons of plugins, and developed in _many different_ ways. If the application gets lock down and cannot persist anything except to some specific location(s), and there is a plugin that by default needs write access somewhere else, you would have to find a way to map anything that needs access to certain locations, if allowed. Worst case, the application goes back to allowing `host` / `home` if there is no work-around.

Don't get me wrong, we can try this way. I think it might be possible, I just being cautious. Would you like to be in the maintainers pool with me :laughing: because I have a feeling this is going to be a full-time job :wink:

Moreover, people might be using this in different ways right now. I think this is doable in the end, there should be a path to not break those folks, leaving some period for the customization. Probably putting this in a beta channel or something like that.

Reply:
I understand this possible issue. I'm ok if you want to put this changes in a beta channel.
Yes I can help in maintaining the flatpak.

Reply:
Let me try to find out if there is a way to put a disclaimer message (or something like that) to announce the changes **only** for the Flatpak. That can be done in the actual one and we should be able to introduce this change some time in the future.